### PR TITLE
nethttp: allow overriding operation name

### LINKED ISSUE #4: Allow operation name to be passed to middleware and traced client
Currently both server and client wrappers use `"HTTP "+r.Method` as the operation name, which in practice is very bad. Considering that these instrumentation functions must be called explicitly, I think they should allow passing the operation name, e.g. via functional option pattern.

### PR DESCRIPTION
**Pull request by yurishkuro**
Fixes #4

### CODE REVIEW

**Comment by yurishkuro** [comment]
@dominikh

File: nethttp/client.go [hunk 1]
```diff
@@ -27,6 +27,30 @@ type Transport struct {
 	http.RoundTripper
 }
 
+type clientOptions struct {
+	opName             string
+	disableClientTrace bool
+}
+
+// ClientOption contols the behavior of TraceRequest() function.
```
Comment (dominikh) [inline nethttp/client.go]: "ClientOption controls the behavior of TraceRequest."

File: nethttp/client.go [hunk 2]
```diff
@@ -27,6 +27,30 @@ type Transport struct {
...
+
+// OperationName returns a ClientOption that sets the operation
+// name for the client-side span.
+func OperationName(opName string) ClientOption {
+	return func(options *clientOptions) {
+		options.opName = opName
+	}
+}
+
+// ClientTrace returns a ClientOption that turn on or off
```
Comment (dominikh) [inline nethttp/client.go]: Typos in "turn" and in "httotrace"

**Comment by dominikh** [comment]
Please prefix the commit messages with `nethttp: ` to denote the affected component.

**Comment by yurishkuro** [comment]
> Please prefix the commit messages with nethttp: to denote the affected component.

I recommend using Squash and Merge button on github to squash commits in the other branch. Then the commit message comes from PR title, which I changed accordingly.

**Pull request closed without merging**